ee/src/main/java/org/jboss/as/ee/concurrent/ControlPointUtils.java

```diff
@@ -67,6 +87,24 @@ public static Runnable doWrap(Runnable runnable, ControlPoint controlPoint) {
         }
     }
 
+    public static <T> Callable<T> doWrapOrThrowException(Callable<T> callable, ControlPoint controlPoint) {
+        if (controlPoint == null || callable == null) {
+            return callable;
+        }
+        try {
+            RunResult result = controlPoint.forceBeginRequest();
+            if (result == RunResult.REJECTED) {
+                ROOT_LOGGER.rejectedDueToMaxRequests();
+            }
+            final ControlledCallable controlledCallable = new ControlledCallable(callable, controlPoint);
```

Comment:
Exceptions in this line and below should not be converted to RejectedException, and request must be completed.